File: android-client/app/src/main/java/com/example/azurecommunication/data/models/AcsModels.kt
package com.example.azurecommunication.data.models

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class TokenRequest(
    val displayName: String = "Anonymous"
)

@JsonClass(generateAdapter = true)
data class TokenResponse(
    val userId: String,
    val token: String,
    val expiresOn: String
)

@JsonClass(generateAdapter = true)
data class CreateThreadRequest(
    val userId: String,
    val displayName: String,
    val topic: String = "Chat"
)

@JsonClass(generateAdapter = true)
data class CreateThreadResponse(
    val threadId: String,
    val topic: String
)

@JsonClass(generateAdapter = true)
data class JoinThreadRequest(
    val userId: String,
    val displayName: String
)

@JsonClass(generateAdapter = true)
data class JoinThreadResponse(
    val success: Boolean,
    val threadId: String,
    val userId: String
)

data class AcsUser(
    val userId: String,
    val token: String,
    val displayName: String
)

enum class CommunicationMode {
    CHAT,
    VIDEO
}

File: ios-client/AzureCommunication/AzureCommunication/Data/Models/AcsModels.swift
//
//  AcsModels.swift
//  AzureCommunication
//

import Foundation

// MARK: - API Request/Response Models

struct TokenRequest: Codable {
    let displayName: String
}

struct TokenResponse: Codable {
    let userId: String
    let token: String
    let expiresOn: String
}

struct CreateThreadRequest: Codable {
    let userId: String
    let displayName: String
    let topic: String
}

struct CreateThreadResponse: Codable {
    let threadId: String
    let topic: String
}

struct JoinThreadRequest: Codable {
    let userId: String
    let displayName: String
}

struct JoinThreadResponse: Codable {
    let success: Bool
    let threadId: String
    let userId: String
}

// MARK: - App Models

struct AcsUser: Equatable {
    let userId: String
    let token: String
    let displayName: String
}

enum CommunicationMode: String, CaseIterable {
    case chat
    case video
}
